roles/pul_nomad/tasks/dpul_c.yml

```diff
@@ -0,0 +1,47 @@
+---
+- name: 'pul_nomad_dpulc | create postgresql db users'
+  community.postgresql.postgresql_user:
+    name: '{{ dpul_c_db_user }}'
+    password: '{{ dpul_c_db_password }}'
+    encrypted: true
+    state: 'present'
+  delegate_to: '{{ dpul_c_postgres_host }}'
+  become: true
+  become_user: '{{ dpul_c_postgres_admin_user }}'
+  run_once: true
+
+- name: 'pul_nomad_dpulc | create postgresql database'
+  community.postgresql.postgresql_db:
+    name: '{{ dpul_c_db_name }}'
+    encoding: 'UTF-8'
+    state: 'present'
+    owner: "{{ dpul_c_db_user }}"
+  delegate_to: '{{ dpul_c_postgres_host }}'
+  become: true
+  become_user: '{{ dpul_c_postgres_admin_user }}'
+  run_once: true
+
+- name: 'pul_nomad_dpulc | ensure access to postgres server for client connections'
+  ansible.builtin.lineinfile:
+    path: '/etc/postgresql/{{ dpul_c_postgres_version }}/main/pg_hba.conf'
+    line: 'host  all      all {{ ansible_default_ipv4.address }}/32       md5'
+  delegate_to: '{{ dpul_c_postgres_host }}'
+  register: remote_postgres_configured
+  throttle: 1 # have this task run one machine at a time to avoid race condition
+  when:
+    - "nomad_node_role == 'client'"
+
+- name: 'pul_nomad_dpulc | reload postgresql'
+  ansible.builtin.service:
+    name: postgresql
+    state: reloaded
+  when:
+    - remote_postgres_configured.changed
+  delegate_to: '{{ dpul_c_postgres_host }}'
+
+- name: 'pul_nomad_dpulc | add the application environment variables'
```

Comment:
Where does this task run? on the VM? On the nomad-server? On the nomad-client?

Reply:
On one of the nomad servers.